### DISCUSSION TITLE
How to implement poll scm in GitHub actions?

### DISCUSSION THREAD

**Discussion by ukreddy-erwin** [community/community#28848] [Other Feature Feedback, Questions, & Ideas]
In Jenkins, there is a poll scm option.

Which will run the job at a scheduled time only if there is any new change from previous build in source repository.

Is there any similar approach in GitHub actions.There is schedule option as in this [link][1]. But that won't check if any new changes there or not and trigger directly.

  [1]: https://jasonet.co/posts/scheduled-actions/

**Comment by danielmeppiel** [comment]
Hello! 

There is no native functionality to allow for a workflow being triggered on polling only when a repo has had changes compared to the previous build. What I would suggest is using a scheduled trigger and then building your own GitHub Action or finding an equivalent in the Marketplace to perform this logic as an initial step. 

Some Actions I could find in the Marketplace that could potentially help detect if there were changes:

https://github.com/marketplace/actions/changed-files
https://github.com/RasaHQ/pr-changed-files-filter

Out of curiosity, could you also explain what's the use case or need behind this?

**Comment by ukreddy-erwin** [comment]
It is not about polling.
The feature in Jenkins is called as pollscm.

To understand it clearly.

Need to setup a schedule to run the action or workflow only in the early morning 4am that too if any new commits added when compared to previous workflow run.if no changes, then it should skip and check again next day

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by danielmeppiel** [reply]
As I said before, you can implement this in GitHub with a scheduled trigger at 4AM, and then use one of the above actions I linked to work on implementing your conditional logic. You can use the [changed-files Action](https://github.com/marketplace/actions/changed-files) and  skip the rest of the workflow if that action detects no changed files.

Hope that helps!

**Reply by ukreddy-erwin** [reply]
It seems.it will.compare the last two commits. But that is not the case.
Let's say day before yesterday commitZ added.
Let's say yesterday commitA added.
Today no commits. Then when build started today, as per your action it will compare A and Z. But it should skip as no new commits. Commit A is already used yesterday

**Reply by danielmeppiel** [reply]
@ukreddy-erwin if you need such a custom behavior and it is not satisfied by open-source actions out there, the best way forward would be to script your own GitHub Action. You can read more about creating actions [here](https://docs.github.com/en/actions/creating-actions).

**Comment by ukreddy-erwin** [comment]
Is there any variable where I can get previous run commit Id of workflow?
So that can write a condition to compare with current workflow run and skip the workflow.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by esmeg02** [comment]
[@Shu](https://malloftoys.com/collections/shu) yeah, it has happened earlier too.

**Reply by ukreddy-erwin** [reply]
Can you share that if you done  similar thing before.
